vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp

```diff
@@ -272,37 +274,163 @@ void ScrollViewComponentView::OnPointerDown(const winrt::Windows::UI::Input::Poi
+  pt.x -= m_scrollVisual.ScrollPosition().x;
+  pt.y -= m_scrollVisual.ScrollPosition().y;
+  return Super::ScrollWheel(pt, delta);
+}
 
-  if (std::any_of(m_children.rbegin(), m_children.rend(), [ptContent, delta](auto child) {
-        return const_cast<CompositionBaseComponentView *>(static_cast<const CompositionBaseComponentView *>(child))
-            ->ScrollWheel(ptContent, delta);
-      }))
-    return true;
+void ScrollViewComponentView::onKeyDown(
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyboardSource &source,
+    const winrt::Microsoft::ReactNative::Composition::Input::KeyRoutedEventArgs &args) noexcept {
+  switch (args.Key()) {
+    case winrt::Windows::System::VirtualKey::End:
+      args.Handled(scrollToEnd(false));
+      break;
+    case winrt::Windows::System::VirtualKey::Home:
+      args.Handled(scrollToStart(false));
+      break;
+    case winrt::Windows::System::VirtualKey::PageDown:
+      args.Handled(pageDown(false));
+      break;
+    case winrt::Windows::System::VirtualKey::PageUp:
+      args.Handled(pageUp(false));
+      break;
+    case winrt::Windows::System::VirtualKey::Up:
+      args.Handled(lineUp(false));
+      break;
+    case winrt::Windows::System::VirtualKey::Down:
+      args.Handled(lineDown(false));
+      break;
+    case winrt::Windows::System::VirtualKey::Left:
+      args.Handled(lineLeft(false));
+      break;
+    case winrt::Windows::System::VirtualKey::Right:
+      args.Handled(lineRight(false));
+      break;
+  }
+}
 
-  if (ptViewport.x >= 0 && ptViewport.x <= m_layoutMetrics.frame.size.width && ptViewport.y >= 0 &&
-      ptViewport.y <= m_layoutMetrics.frame.size.height) {
-    m_scrollVisual.ScrollBy({0, static_cast<float>(-delta), 0});
-    return true;
+bool ScrollViewComponentView::scrollToEnd(bool animate) noexcept {
+  if ((((m_contentSize.height - m_layoutMetrics.frame.size.height) * m_layoutMetrics.pointScaleFactor) -
+       m_scrollVisual.ScrollPosition().y) < 1.0f) {
+    return false;
   }
 
-  return false;
+  auto x = (m_contentSize.width - m_layoutMetrics.frame.size.width) * m_layoutMetrics.pointScaleFactor;
+  auto y = (m_contentSize.height - m_layoutMetrics.frame.size.height) * m_layoutMetrics.pointScaleFactor;
+  m_scrollVisual.TryUpdatePosition({static_cast<float>(x), static_cast<float>(y), 0.0f}, animate);
+  return true;
+}
+
+bool ScrollViewComponentView::scrollToStart(bool animate) noexcept {
+  m_scrollVisual.TryUpdatePosition({0.0f, 0.0f, 0.0f}, animate);
+  return true;
+}
+
+bool ScrollViewComponentView::pageUp(bool animate) noexcept {
+  if (m_scrollVisual.ScrollPosition().y <= 0.0f) {
+    return false;
+  }
+
+  float scrollToVertical =
+      m_scrollVisual.ScrollPosition().y - (m_layoutMetrics.frame.size.height * m_layoutMetrics.pointScaleFactor);
+  float scrollToHorizontal = m_scrollVisual.ScrollPosition().x;
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
```

Comment:
Why isn't `animate` passed through?
Plenty more below. Same feedback for all.

Reply:
reworked the code to animate keyboarding - which seems to align with XAML's default behavior.
I also reworked the animate scroll behavior to alter the duration of the animation to be based on the distance to travel, also aligning more with XAML's behavior.